crates/solver/src/driver.rs

```diff
@@ -178,18 +180,31 @@ impl Driver {
             .await
             .context("error retrieving current auction")?;
 
+        // It doesn't make sense to solve the same auction again because we wouldn't be able to
+        // store competition info etc.
+        if matches!(self.last_attempted_settlement, Some(id) if id == auction.id) {
```

Comment:
nit: 👇  seems a little more concise
```suggestion
        if self.last_attempted_settlement == Some(auction.id) {
```